pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -90,33 +95,33 @@ func (r *ReconcileServiceDiscovery) Reconcile(request reconcile.Request) (reconc
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			daemonSet := &appsv1.DaemonSet{}
+			deployment := &appsv1.Deployment{}
 			opts := []client.DeleteAllOfOption{
 				client.InNamespace(request.NamespacedName.Namespace),
-				client.MatchingLabels{"app": appName},
+				client.MatchingLabels{"deployment": deploymentName},
 			}
-			err := r.client.DeleteAllOf(context.TODO(), daemonSet, opts...)
+			err := r.client.DeleteAllOf(context.TODO(), deployment, opts...)
 			return reconcile.Result{}, err
 		}
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
 	lightHouseAgent := newLighthouseAgent(instance)
-	if _, err = helpers.ReconcileDaemonSet(instance, lightHouseAgent, reqLogger,
+	if _, err = helpers.ReconcileDeployment(instance, lightHouseAgent, reqLogger,
 		r.client, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 	return reconcile.Result{}, nil
 }
 
-func newLighthouseAgent(cr *submarinerv1alpha1.ServiceDiscovery) *appsv1.DaemonSet {
+func newLighthouseAgent(cr *submarinerv1alpha1.ServiceDiscovery) *appsv1.Deployment {
+	replicas := int32(1)
 	labels := map[string]string{
 		"app":       "submariner-lighthouse-agent",
 		"component": "lighthouse-agent",
 	}
-
 	matchLabels := map[string]string{
-		"app": appName,
+		"deployment": deploymentName,
```

Comment:
Why was this changed from "app" to "deployment"? I guess it doesn't matter either way but we use "app" in other places and it seems more standard.

Reply:
Also since you've made a `deploymentName` const, shouldn't we use that on L 114 and L144 as well?

Reply:
Done.